prow/plugins/approve/approve_test.go

```diff
@@ -729,6 +729,7 @@ Approvers can cancel approval by writing ` + "`/approve cancel`" + ` in a commen
 			reviews: []github.Review{
 				newTestReview("cjwagner", "stuff", "COMMENTED"),
 				newTestReview("cjwagner", "unsubmitted stuff", "PENDING"),
+				newTestReview("cjwagner", "dismissed stuff", "DISMISSED"),
```

Comment:
Need a test case that transitions from approved --> non-approved via dismissal

Reply:
I'm not sure how to test this case. When a review is dismissed, its state is updated in place, and the table tests don't support this AFAICT. Can you suggest a testing strategy @stevekuznetsov?

Reply:
Oh I see, nevermind then

Reply:
:smile: I noticed that no case ensured a dismissed review doesn't cancel a separate approval by the same user, so I added one.